rust/extns/numaflow-sqs/src/lib.rs

```diff
@@ -0,0 +1,110 @@
+/// Library for robust SQS message handling using an actor-based architecture.
+///
+/// This module provides a fault-tolerant interface for interacting with Amazon SQS,
+/// with a focus on:
+/// - Error propagation and handling for AWS SDK errors
+/// - Actor-based concurrency model for thread safety
+/// - Clean abstraction of SQS operations
+use aws_sdk_sqs::error::SdkError;
+use tokio::sync::oneshot;
+pub mod source;
+
+/// Custom error types for the SQS client library.
+///
+/// Design goals:
+/// - Ergonomic error handling with thiserror
+/// - Clear error propagation from AWS SDK
+/// - Explicit handling of actor communication failures
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Failed with SQS error - {0}")]
+    SQS(aws_sdk_sqs::Error),
```

Comment:
to handle this error in the core, we will have to import aws_sdk_sqs, right? so convert this to numaflow error.